test/src/test/java/hudson/util/BootFailureTest.java

```diff
@@ -154,6 +154,8 @@ public File allocate() throws Exception {
         @Override
         public void onLoaded() {
             wa.contextDestroyed(null);
+            // make the Jenkins.<init> thread abort
+            throw new Error();
```

Comment:
Not going to work once we fix `ItemListener` to catch errors from `onLoaded` implementations, though that is also true of `InduceBootFailure`.

But the :bug: is because the fact you needed to do this suggests to me that you somehow regressed the logic in `WebAppMain.contextDestroyed` which is supposed to be interrupting the `initThread` and which this test (from #1950) was verifying.